### DISCUSSION TITLE
[IDEA] Can you explain what are khoj advantages over just using Claude MCP with its integration capabilities?

### DISCUSSION THREAD

**Discussion by shlomitsur** [khoj-ai/khoj#1022] [Q&A]
hi.

I mean no offence I just want to know what is the added value of using Khoj.
Thanks!

**Comment by vlazic** [comment]
Good question!

As someone who has recently started experimenting locally with Khoj and MCP, I was just about to ask the development team if they are planning to integrate MCP. In the long term, it seems more worthwhile to rely on MCP rather than develop and support one's own solutions.

I think it would be a valuable addition, where they could delegate the creation of tools to others and provide greater freedom to users.

Now that we have MCP, I believe there is significant room for MCP clients.
I think Khoj already has a head start in this area and could focus on user experience, the RAG component, and other features.

**Comment by debanjum** [comment]
Good question indeed! How are you using MCP @shlomitsur, @vlazic? I haven't used it yet myself but it sounds super interesting.

Given MCP is an open protocol for interacting with content, support for it can be added to Khoj. At which point the advantages of Khoj remain the same; use with any AI model, better answers, self-hostable, transparent/open-source/community-built and all the rest of Khoj capabilities (e.g agents, automations, research mode).

> I was just about to ask the development team if they are planning to integrate MCP. In the long term, it seems more worthwhile to rely on MCP rather than develop and support one's own solutions.

I've to look into exactly what is required to use MCP in Khoj and what are the concrete advantages to do so. But discussions like this are useful to understand what folks want and for what specific use-cases. So if/when we do build the integration, we have a concrete set of capabilities to prioritize

**Comment by vlazic** [comment]
> Good question indeed! How are you using MCP @shlomitsur, @vlazic? I haven't used it yet myself but it sounds super interesting.

Few days ago [Cline](https://github.com/cline/cline) added MCP integration so I use it through that. Unfortunately, I'm not yet using it via Claude Desktop (Linux is not supported), but I would love to use it there as well. As soon as MCP appeared, the community began to [create various integrations](https://github.com/punkpeye/awesome-mcp-servers), the entire ecosystem is growing.

> Given MCP is an open protocol for interacting with content, support for it can be added to Khoj. At which point the advantages of Khoj remain the same; use with any AI model, better answers, self-hostable, transparent/open-source/community-built and all the rest of Khoj capabilities (e.g agents, automations, research mode).

Exactly

> I've to look into exactly what is required to use MCP in Khoj and what are the concrete advantages to do so. But discussions like this are useful to understand what folks want and for what specific use-cases. So if/when we do build the integration, we have a concrete set of capabilities to prioritize

I'm particularly interested in Khoj because of how well it works with Obsidian. I find it performs better than other similar Obsidian plugins, and I appreciate that it runs as a local server. Overall, I think it has great potential.

As for the MCP, I currently use it through Cline to automatically generate Markdown files in a specific folder in my Obsidian vault. I also plan to regularly import data from some third-party services, use an LLM to format it according to specific Obsidian templates, and save it as markdown files in my vault. For this purpose, I plan to create my own MCP integration (unless someone else develops it first).

Second example: Khoj already supports internet search but uses a specific API. I can only use services that Khoj has chosen, and I need to pay a monthly subscription. I might rarely need these features, and I would like to choose which API I use for that purpose.

**Comment by debanjum** [comment]
I'll convert this issue into a discussion (as don't want to edit original title, question). But created a [corresponding issue](https://github.com/khoj-ai/khoj/issues/1023) to evaluate Claude MCP integration with Khoj
